### PR TITLE
Clarified the use of Signable in metadata.py

### PR DESCRIPTION
**Pull request by DarikshaAnsari**
**Fixes issue #621**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by DarikshaAnsari** [comment]
@lukpueh I've resolved the issue with Envelope.from_signable to clarify its usage and avoid the misconception about requiring subclassing of Signable.

File: in_toto/models/metadata.py [hunk 1]
```diff
@@ -146,13 +146,36 @@ class Envelope(SSlibEnvelope, Metadata):
...
+        to DSSE's design principles. DSSE aims to avoid the complexities
+        and potential issues of canonicalization by using a straightforward
+        JSON serialization approach.
+
+        While this method currently expects an instance of `Signable` for
+        compatibility reasons, it is the non-canonical serialization that
+        is crucial for DSSE envelopes. The payloads do not strictly need
+        to subclass `Signable`, but they must be serializable to JSON.
+
+        Parameters:
```
Comment (lukpueh) [inline in_toto/models/metadata.py]: Consistency nit: 
```suggestion
        Arguments:
```

File: in_toto/models/metadata.py [hunk 2]
```diff
@@ -146,13 +146,36 @@ class Envelope(SSlibEnvelope, Metadata):
...
+        signable (Signable): The object to be signed. This object is expected
+        to be compatible with `Signable`, but its `signable_bytes` property
+        will not be used.
+
+        Returns:
+        Envelope: An envelope containing the serialized signable object.
```
Comment (lukpueh) [inline in_toto/models/metadata.py]:
```suggestion
            signable (Signable): The object to be signed. This object is expected
                to be compatible with `Signable`, but its `signable_bytes` property
                will not be used.

        Returns:
            Envelope: An envelope containing the serialized signable object.
```

File: in_toto/models/metadata.py [hunk 3]
```diff
@@ -146,13 +146,36 @@ class Envelope(SSlibEnvelope, Metadata):
...
+        This method intentionally bypasses the `signable_bytes` property of
+        the `Signable` class to avoid canonicalization, which is contrary
+        to DSSE's design principles. DSSE aims to avoid the complexities
+        and potential issues of canonicalization by using a straightforward
+        JSON serialization approach.
+
+        While this method currently expects an instance of `Signable` for
+        compatibility reasons, it is the non-canonical serialization that
+        is crucial for DSSE envelopes. The payloads do not strictly need
+        to subclass `Signable`, but they must be serializable to JSON.
```
Comment (lukpueh) [inline in_toto/models/metadata.py]: This is great, but maybe a bit wordy. Would you mind condensing it a little? The key points we want to convey are:

- The method takes a `Signable`, only because it is a base class for `Link` and `Layout`.
- `Signable.signable_bytes` is ignored, because canonical json is is undesired in the DSSE context.
-  Instead, this method serialises the `Signable` using regular JSON.

**Review by lukpueh — CHANGES_REQUESTED** [review]
Thanks for the patch, @DarikshaAnsari!

**Comment by DarikshaAnsari** [comment]
@lukpueh PTAL!!

File: in_toto/models/metadata.py [hunk 4]
```diff
@@ -146,13 +146,31 @@ class Envelope(SSlibEnvelope, Metadata):
 
     @classmethod
     def from_signable(cls, signable: Signable) -> "Envelope":
-        """Creates DSSE envelope with signable bytes as payload."""
+        """Creates DSSE envelope with signable bytes as payload.
 
+        Note:
+        The method accepts a `Signable` because it serves as a base class
+        for `Link` and `Layout`.
+        This method intentionally avoid using `Signable.signable_bytes` to
```
Comment (lukpueh) [inline in_toto/models/metadata.py]:
```suggestion
        This method intentionally avoids using `Signable.signable_bytes` to
```

**Review by lukpueh — APPROVED** [review]
